### PR TITLE
Changed the Typing for Case Component

### PR DESCRIPTION
**Pull request by louiss0**
The point of this pull request is for you to change the types for `<AstroCase/>`. The user is not supposed to be allowed to pass of and default at the same time. When it comes to the use of types it's better to just be more explicit about what types are supposed to be being passed in. 

I have decided to use pnpm for this project. It's not easy to test packages without it. pnpm allows you to use a local package without all of the problems with npm. I replaced the demo package with the one that is used in this repo. If you don't want to use it that's fine but if you do then it will be so much more useful than npm.

### CODE REVIEW

File: demo/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "serve": "astro preview"
   },
   "devDependencies": {
-	"@astropub/flow": "0.3.6",
-    "astro": "latest"
+    "astro": "latest",
+	"@astropub/flow": "workspace:*0.3.6"
```
Comment (louiss0) [inline demo/package.json]: This comes from the local version that is in the repo

File: packages/flow/lib/AstroCase.d.ts [hunk 2]
```diff
@@ -1,21 +1,17 @@
...
-		}
-	: never
-: 'default' extends keyof T
-	? {
-		default: "" | true
-	}
-: any
 
-export default function Case<T>(props: Props<T>): any
+type AllowedTypes = string | null | bigint | Record<PropertyKey, unknown> | undefined | number | boolean
```
Comment (louiss0) [inline packages/flow/lib/AstroCase.d.ts:2]: If you don't know `Record<PropertyKey, unknown>` stands for the typical object literals that we type every day. The point of allowed types is to just be explicit. when it comes to writing an object this is the right way to express this intent.

File: packages/flow/lib/AstroCase.d.ts [hunk 3]
```diff
@@ -1,21 +1,17 @@
...
+type FilledCase = {
+    of: AllowedTypes |  ((value:unknown)=> boolean);
+    default?:never
+}
+
+type DefaultCase = {
+    default:true;
+    of?:never
+    }
+type Case =  FilledCase |  DefaultCase
```
Comment (louiss0) [inline packages/flow/lib/AstroCase.d.ts:13]: This typing ensures that the user can only choose between the `of=` or the `default=` attribute when it comes to typing a `<AstroCase/>`

File: packages/flow/lib/AstroSwitch.d.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
+
 export interface Props {
-	of: any
+	of: unknown
```
Comment (louiss0) [inline packages/flow/lib/AstroSwitch.d.ts:3]: I think `unknown` is better than any

File: packages/flow/lib/AstroWhen.d.ts [hunk 5]
```diff
@@ -1,3 +1,4 @@
-export type Props = Record<any, any>
+
+export type Props = Record<string, any>
```
Comment (louiss0) [inline packages/flow/lib/AstroWhen.d.ts:2]: I think any was a lazy thing to do but an attribute key is a `string`

**Comment by louiss0** [comment]
Please don't forget about this request

**Comment by jonathantneal** [comment]
I’ll probably merge this, but then roll back some of the pnpm stuff. Having the lock files in here is too messy. If we can find a way to support both, they can stay. Otherwise, tho, it’s starting to look too engineered.

**Comment by louiss0** [comment]
Did you publish the changes to npm yet?

**Pull request merged**